tests/sentry/integrations/vercel/test_integration.py

```diff
@@ -118,3 +122,212 @@ def test_use_existing_installation(self):
+        )
+        # mock project secret
+        responses.add(
+            responses.GET,
+            "https://api.vercel.com/v3/now/secrets/%s" % "sentry_project_1",
+            body=ApiError('The secret "%s" was not found.' % "sentry_project_1", code=404),
+        )
+        responses.add(
+            responses.POST, "https://api.vercel.com/v2/now/secrets", json={"uid": "sec_456"},
+        )
+        # mock DSN secret
+        responses.add(
+            responses.GET,
+            "https://api.vercel.com/v3/now/secrets/%s" % "next_public_sentry_dsn_1",
+            body=ApiError('The secret "%s" was not found.' % "next_public_sentry_dsn_1", code=404),
+        )
+
+        responses.add(
+            responses.POST, "https://api.vercel.com/v2/now/secrets", json={"uid": "sec_789"},
+        )
+        # mock get envs for all
+        responses.add(
+            responses.GET,
+            "https://api.vercel.com/v5/projects/%s/env"
+            % "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H",
+            json={"envs": [],},
+        )
+        # mock org env var
+        responses.add(
+            responses.POST,
+            "https://api.vercel.com/v4/projects/%s/env"
+            % "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H",
+            json={"value": "sec_123", "target": "production", "key": "sentry_org",},
+        )
+        # mock project env var
+        responses.add(
+            responses.POST,
+            "https://api.vercel.com/v4/projects/%s/env"
+            % "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H",
+            json={"value": "sec_456", "target": "production", "key": "sentry_project_1",},
+        )
+        # mock dsn env var
+        responses.add(
+            responses.POST,
+            "https://api.vercel.com/v4/projects/%s/env"
+            % "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H",
+            json={"value": "sec_789", "target": "production", "key": "next_public_sentry_dsn_1",},
+        )
+
+        org = self.organization
+        data = {"project_mappings": [[1, "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H"],]}
+        enabled_dsn = ProjectKey.get_default(project=Project.objects.get(id=1)).get_dsn(public=True)
+        integration = Integration.objects.get(provider=self.provider.key)
+        installation = integration.get_installation(org.id)
+        org_integration = OrganizationIntegration.objects.get(
+            organization_id=org.id, integration_id=integration.id
+        )
+        assert org_integration.config == {}
+        installation.update_organization_config(data)
+        org_integration = OrganizationIntegration.objects.get(
+            organization_id=org.id, integration_id=integration.id
+        )
+        assert org_integration.config == {
+            "project_mappings": [[1, "Qme9NXBpguaRxcXssZ1NWHVaM98MAL6PHDXUs1jPrgiM8H"],]
+        }
+
+        assert responses.calls[4].response.code == 404  # check org secret doesnt exist
```

Comment:
@ceorourke No need to check if it gets a 404 if the mock is setting the 404. Same with the 200s below/